packages/glimmer-runtime/lib/dom/props.ts

```diff
@@ -0,0 +1,77 @@
+export function isAttrRemovalValue(value) {
+  return value === null || value === undefined;
+}
+/*
+ *
+ * @method normalizeProperty
+ * @param element {HTMLElement}
+ * @param slotName {String}
+ * @returns {Object} { name, type }
+ */
+export function normalizeProperty(element, slotName) {
+  var type, normalized;
+
+  if (slotName in element) {
+    normalized = slotName;
+    type = 'prop';
+  } else {
+    var lower = slotName.toLowerCase();
+    if (lower in element) {
+      type = 'prop';
+      normalized = lower;
+    } else {
+      type = 'attr';
+      normalized = slotName;
+    }
+  }
+
+  if (type === 'prop' &&
+      (normalized.toLowerCase() === 'style' ||
+       preferAttr(element.tagName, normalized))) {
+    type = 'attr';
+  }
+
+  return { normalized, type };
+}
+
+// properties that MUST be set as attributes, due to:
+// * browser bug
+// * strange spec outlier
+const ATTR_OVERRIDES = {
+
+  // phantomjs < 2.0 lets you set it as a prop but won't reflect it
+  // back to the attribute. button.getAttribute('type') === null
+  BUTTON: { type: true, form: true },
+
+  INPUT: {
+    // TODO: remove when IE8 is droped
+    // Some versions of IE (IE8) throw an exception when setting
```

Comment:
IE8 is unsupported. IE9 is the minimum version afaik (and is definitely the lowest tested version of IE)